BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -348,7 +351,7 @@ public Part GetPart()
                     if (weapon.Current.GetShortName() != GetShortName()) continue;
                     if (weapon.Current.AmmoID != AmmoID && weapon.Current.AmmoID != lastAmmoID)
                     {
-                        vessel.GetConnectedResourceTotals(weapon.Current.AmmoID, out double ammoCurrent, out double ammoMax);
+                        GetAmmoCount(out double ammoCurrent, out double ammoMax);
```

Comment:
This is going to break ammo count for any heterogeneous weapon group using different ammos, as GetAmmoCount is only going to return this weapon's ammo, not weapon.Current.AmmoID. GetAmmoCount will need to be modified to also take a resourceID int, and use that for the part.GetConnectedResourceTotals(...)

Reply:
Fixed, good catch, I missed that case being different.